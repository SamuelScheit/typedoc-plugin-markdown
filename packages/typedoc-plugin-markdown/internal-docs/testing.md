---
title: Testing guide
description: Guidance of running tests and creating fixtures.
---

# Testing

## Running tests

```bash
npm run test
```

If you require to update the snapshots run the following command:

```bash
npm run test:update
```

## Test Fixtures

A series of snapshot tests are used to ensure the output of the plugin is consistent.

- Example code for fixtures are located at `/test/fixtures/src`.
- Fixtures output scenarios are setup using a config file (`/test/fixtures/config.ts`) that provides a mechanism to easily provide a combination of options.
