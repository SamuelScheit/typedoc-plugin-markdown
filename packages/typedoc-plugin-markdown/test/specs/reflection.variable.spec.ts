import { expectFileToEqual } from '@devtools/testing';

describe(`Variable Reflection`, () => {
  test(`should compile variable assigned to a string`, () => {
    expectFileToEqual('reflections', 'members', 'variables/stringVariable.md');
  });

  test(`should compile variable assigned to an object literal`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      'variables/objectLiteralVariable.md',
    );
  });

  test(`should compile type operator variable`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      'variables/typeOperatorVariable.md',
    );
  });

  test(`should compile object with symbol`, () => {
    expectFileToEqual(
      'reflections',
      'members',
      'variables/objectWithSymbol.md',
    );
  });
});
