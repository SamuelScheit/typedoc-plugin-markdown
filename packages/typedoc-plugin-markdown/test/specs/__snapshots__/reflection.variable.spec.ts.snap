// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Variable Reflection should compile object with symbol: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Variable: objectWithSymbol

> **\`const\`** **objectWithSymbol**: \`object\`

Comments variable with symbol

## Type declaration

### [sym]

> **[sym]**: \`string\` = \`'value'\`

Comments for symbol

## Source

[variables.ts:51](http://source-url)
"
`;

exports[`Variable Reflection should compile object with symbol: (Output File Strategy "members") (Option Group "2") 1`] = `
"# objectWithSymbol

\`\`\`ts
const objectWithSymbol: object;
\`\`\`

Comments variable with symbol

## Type declaration

| Member | Type | Value | Description |
| :------ | :------ | :------ | :------ |
| \`[sym]\` | \`string\` | 'value' | Comments for symbol |

## Source

[variables.ts:51](http://source-url)
"
`;

exports[`Variable Reflection should compile type operator variable: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Variable: typeOperatorVariable

> **\`const\`** **typeOperatorVariable**: unique \`symbol\`

Comments for typeOperatorVariable

## Source

[variables.ts:13](http://source-url)
"
`;

exports[`Variable Reflection should compile type operator variable: (Output File Strategy "members") (Option Group "2") 1`] = `
"# typeOperatorVariable

\`\`\`ts
const typeOperatorVariable: unique symbol;
\`\`\`

Comments for typeOperatorVariable

## Source

[variables.ts:13](http://source-url)
"
`;

exports[`Variable Reflection should compile variable assigned to a string: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Variable: stringVariable

> **\`const\`** **stringVariable**: \`"stringConstWithDefaultValue"\` = \`'stringConstWithDefaultValue'\`

A simple string variable

## Source

[variables.ts:8](http://source-url)
"
`;

exports[`Variable Reflection should compile variable assigned to a string: (Output File Strategy "members") (Option Group "2") 1`] = `
"# stringVariable

\`\`\`ts
const stringVariable: "stringConstWithDefaultValue" = 'stringConstWithDefaultValue';
\`\`\`

A simple string variable

## Source

[variables.ts:8](http://source-url)
"
`;

exports[`Variable Reflection should compile variable assigned to an object literal: (Output File Strategy "members") (Option Group "1") 1`] = `
"# Variable: objectLiteralVariable

> **\`const\`** **objectLiteralVariable**: \`object\`

Comments for objectLiteralVariable

## Type declaration

### \\*

> **\\***: \`string\` = \`'*'\`

### \\*\\*foo\\*\\*

> **\\*\\*foo\\*\\***: \`string\` = \`'**foo**'\`

### \\*foo\\*

> **\\*foo\\***: \`string\` = \`'*foo*'\`

### \\<

> **\\<**: \`string\` = \`'<'\`

### \\<foo

> **\\<foo**: \`string\` = \`'<foo'\`

### \\<foo\\>

> **\\<foo\\>**: \`string\` = \`'<foo>'\`

### \\>

> **\\>**: \`string\` = \`'>'\`

### \\_

> **\\_**: \`string\` = \`'_'\`

### \\_foo\\_

> **\\_foo\\_**: \`string\` = \`'_foo_'\`

### prop\\_with\\_underscore

> **prop\\_with\\_underscore**: \`string\` = \`'prop_with_underscore'\`

### prop\\|with\\|pipes

> **prop\\|with\\|pipes**: \`string\` = \`'prop|with|pipes'\`

### valueA

> **valueA**: \`number\` = \`100\`

### valueB

> **valueB**: \`boolean\` = \`true\`

### valueX

> **valueX**: \`object\`

### valueX.valueA

> **valueA**: \`number\`[]

### valueX.valueY()

> **valueY**: (\`z\`) => \`object\`

#### Parameters

• **z**: \`string\`

#### Returns

\`object\`

##### a

> **a**: \`string\` = \`'test'\`

##### b

> **b**: \`string\` = \`z\`

##### c

> **c**: \`object\`

##### c.a

> **a**: \`number\` = \`1\`

##### c.b

> **b**: \`number\` = \`2\`

### valueX.valueZ

> **valueZ**: \`string\` = \`'foo'\`

### valueY()

> **valueY**: (\`unionParam\`, \`_undercoreParam_\`) => \`string\`

#### Parameters

• **unionParam**: \`"a"\` \\| \`"b"\`

• **\\_undercoreParam\\_**: \`string\`

#### Returns

\`string\`

### \\|

> **\\|**: \`string\` = \`'|'\`

### ~

> **~**: \`string\` = \`'~'\`

## Source

[variables.ts:18](http://source-url)
"
`;

exports[`Variable Reflection should compile variable assigned to an object literal: (Output File Strategy "members") (Option Group "2") 1`] = `
"# objectLiteralVariable

\`\`\`ts
const objectLiteralVariable: object;
\`\`\`

Comments for objectLiteralVariable

## Type declaration

| Member | Type | Value |
| :------ | :------ | :------ |
| \`*\` | \`string\` | '\\*' |
| \`**foo**\` | \`string\` | '\\*\\*foo\\*\\*' |
| \`*foo*\` | \`string\` | '\\*foo\\*' |
| \`<\` | \`string\` | '\\<' |
| \`<foo\` | \`string\` | '\\<foo' |
| \`<foo>\` | \`string\` | '\\<foo\\>' |
| \`>\` | \`string\` | '\\>' |
| \`_\` | \`string\` | '\\_' |
| \`_foo_\` | \`string\` | '\\_foo\\_' |
| \`prop_with_underscore\` | \`string\` | 'prop\\_with\\_underscore' |
| prop\\|with\\|pipes | \`string\` | 'prop\\|with\\|pipes' |
| \`valueA\` | \`number\` | 100 |
| \`valueB\` | \`boolean\` | true |
| \`valueX\` | \`object\` | - |
| \`valueX.valueA\` | \`number\`[] | - |
| \`valueX.valueY\` | (\`z\`: \`string\`) => \`object\` | - |
| \`valueX.valueZ\` | \`string\` | 'foo' |
| \`valueY\` | (\`unionParam\`: \`"a"\` \\| \`"b"\`, \`_undercoreParam_\`: \`string\`) => \`string\` | - |
| \\| | \`string\` | '\\|' |
| \`~\` | \`string\` | '~' |

## Source

[variables.ts:18](http://source-url)
"
`;
